### PR TITLE
planning: more fixes/improvements in emergency_pull_over scenario

### PR DESCRIPTION
**Pull request by jmtao**

### CODE REVIEW

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc [hunk 1]
```diff
@@ -610,7 +619,8 @@ bool PathBoundsDecider::FindEmergencyPullOverS(
   const double adc_end_s = reference_line_info.AdcSlBoundary().end_s();
 
   // TODO(all): to be implemented
-  *pull_over_s = adc_end_s + 10.0;
+
+  *pull_over_s = adc_end_s + 15.0;
```
Comment (Capri2014) [inline modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: should this `15` be a config?

File: modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc [hunk 2]
```diff
@@ -101,16 +101,30 @@ common::Status PiecewiseJerkPathOptimizer::Process(
...
+      double pull_over_y;
       if (pull_over_status.has_position() &&
           pull_over_status.position().has_x() &&
-          pull_over_status.position().has_y() &&
+          pull_over_status.position().has_y()) {
+        pull_over_x = pull_over_status.position().x();
+        pull_over_y = pull_over_status.position().y();
+        pull_over = true;
+      } else if (emergency_pull_over_status.has_position() &&
+          emergency_pull_over_status.position().has_x() &&
```
Comment (Capri2014) [inline modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc]: Do we also want to add the `heading diff` and `IoU` check here as well?

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc (hunk 1 above)
Comment (jmtao) [reply modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: yes. there's a TODO tag and will be implemented soon

File: modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc]: need look into this, and add later if needed.

**Pull request merged**